src/collector/Backend.cpp

```diff
@@ -73,7 +73,7 @@ void Backend::update(const BackendStat & stat)
     double ts2 = double(stat.get_timestamp()) / 1000000.0;
     double d_ts = ts2 - ts1;
 
-    if (d_ts > 1.0) {
+    if (d_ts > 1.0 && !stat.error) {
```

Comment:
Can you put comments on this with explanation how it smooth calculation and why it won't lead to dropping all updates?